avocado/core/sysinfo.py

```diff
@@ -614,10 +647,14 @@ def start_test_hook(self):
         if self.log_packages:
             self._log_installed_packages(self.pre_dir)
 
-    def end_test_hook(self):
+    def end_test_hook(self, status=""):
         """
         Logging hook called after a test finishes.
         """
+        if status == "FAIL":
+            for log_hook in self.end_test_fail_collectibles:
+                log_hook.run(self.post_dir)
+            return
```

Comment:
I guess it is better to keep the default collectibles run for all status and additionally fail_collectibles  when test failed.

Reply:
@sathnaga that is what I am doing here, right ?

Reply:
lets not restrict the default run, just add our hook when enabled.

Now that we have taken out common files, both collection are needed.

Reply:
@sathnaga In my implementation, the hook gets called twice, once for all cases, and one for fail cases.
In all cases, it runs for all generic collectibles.
In fail cases, it runs for fail collectibles alone.

Can you please explain your thoughts a bit more ?

Reply:
that's what am trying to understand, why two times to call, is this not suffice, https://github.com/avocado-framework/avocado/blob/214d2191d5035647c9aa4370f888923f16819466/avocado/core/test.py#L950, just that we need export the status, or use global status variable if any?

Reply:
@sathnaga Right. As of today, sysinfo does not have the context of status at all. And from the existing sysinfo hook call, status does not exist. So, I had to take this approach of explicitly calling the hook with status in fail(), by not disturbing any of the existing workflow.